ai-service/src/ai_service/embeddings/tests/test_encoding.py

```
@@ -0,0 +1,130 @@
+    """Test input validation and error handling."""
+
+    @pytest.mark.parametrize("invalid_input", ["", "   ", "\t", "\n"])
+    def test_rejects_empty_strings(self, invalid_input: str):
+        with pytest.raises(
+            errors.EmbeddingError,
+            match="Cannot embed empty or whitespace-only texts",
+        ):
+            encoding.embed_query(invalid_input)
+            encoding.embed_documents([invalid_input])
+
+    def test_embed_documents_rejects_empty_list(self):
```

Ruff (0.12.2)

Missing return type annotation for public function `test_embed_documents_rejects_empty_list`

Add return type annotation: `None`

(ANN201)